Cleaned up getting-started docs, added more docs for Querier. (#1541)
thanos-io/thanos: docs/quick-tutorial.md

We will be progressively updating our Katacoda Course with more scenarios.

On top of this feel free to go through our tutorial presented here:

### Prometheus

Thanos is based on Prometheus. With Thanos you use more or less Prometheus features depending on the deployment model, however
Prometheus always stays as integral foundation for *collecting metrics* and alerting using local data.

Thanos bases itself on vanilla [Prometheus](https://prometheus.io/) (v2.2.1+). We plan to support *all* Prometheus version beyond this version.

NOTE: It is highly recommended to use Prometheus 2.13 (available in next Prometheus release) due to Prometheus remote read improvements.

Always make sure to run Prometheus as recommended by Prometheus team, so:

* Put Prometheus in the same failure domain. This means same network, same datacenter as monitoring services.
* Use persistent disk to persist data across Prometheus restarts.
* Use local compaction for longer retentions.
* Do not change min TSDB block durations. 
* Do not scale out Prometheus unless necessary. Single Prometheus is highly efficient (: 

> However if you scale out your Prometheus instance, Thanos is exactly helping with that. 

Reviewer: ```suggestion
We recommend using Thanos when you need to scale out your Prometheus instance.
```

### Components

Following the [KISS](https://en.wikipedia.org/wiki/KISS_principle) and Unix philosophies, Thanos is made of a set of components with each filling a specific role.

> * Sidecar: connects to Prometheus and reads its data for query and/or upload it to cloud storage.

Reviewer: ```suggestion
* Sidecar: connects to Prometheus, reads its data for query and/or uploads it to cloud storage.
```

> * Store Gateway: exposes the content of a cloud storage bucket.

Reviewer: ```suggestion
* Store Gateway: serves metrics inside of a cloud storage bucket.
```

> * Compactor: compacts, downsamples and applies retention on the data stored in remote storage.

Reviewer: ```suggestion
* Compactor: compacts, downsamples and applies retention on the data stored in cloud storage bucket.
```

* Receiver: receives data from Prometheus' remote-write WAL, exposes it and/or upload it to cloud storage.
* Ruler/Rule: evaluates recording and alerting rules against data in Thanos for exposition and/or upload.
* Querier/Query: implements Prometheus' v1 API to aggregate data from the underlying components.

See those components on this diagram:

<img src="img/arch.jpg" class="img-fluid" alt="architecture overview" />

### [Sidecar](components/sidecar.md)

Thanos integrates with existing Prometheus servers through a [Sidecar process](https://docs.microsoft.com/en-us/azure/architecture/patterns/sidecar#solution), which runs on the same machine or in the same pod as the Prometheus server. 

> The purpose of the Sidecar is to backup Prometheus data into an Object Storage bucket, and giving other Thanos components access to the Prometheus instance the Sidecar is attached to via a gRPC API.

Reviewer: ```suggestion
The purpose of the Sidecar is to backup Prometheus data into an Object Storage bucket, and give other Thanos components access to the Prometheus metrics via a gRPC API.
```
